### ISSUE TITLE
Razorpay Checkout not opening in IOS

### ISSUE THREAD

**Issue by varsh8th** [razorpay/react-native-razorpay#345]
## Description

Razorpay checkout not opening in mobile app, but only in ios. 
It works fine in android

## Razorpay SDK Version :
Bundle version string : 1.2.13
react-native-razorpay": "^2.2.7"

## Xcode Version :

Version 12.2 (12B45b)

### What you did:

Tried to use checkout by clicking on a button in the app

### What happened:

Nothing happens, the loader circles a little but other than that the checkout screen never opens.
At least in simulator sometimes it pops up but in production on any device it doesn't open.

## Steps To Reproduce

https://www.loom.com/share/bf4b55f639de403baee817824fea837a

## Code example, screenshot, or link to a repository:

fetch(`https://api.razorpay.com/v1/orders`, {
      method: 'POST',
      headers: new Headers({
        Accept: 'application/json',
        'Content-Type': 'application/json',
        'Authorization': `Bearer Key Here`

      }),
      body: JSON.stringify({
        amount: 100,
        currency: currency,
        payment: {
          capture: "automatic",
          capture_options: {
            refund_speed: "normal"
          }
        },
        transfers: [
          {
            account: linkedAccount,
            amount: 100,
            currency: currency,
            notes: {
              name: '1',
              email: '1'
            },
            linked_account_notes: [name , email]
          },
        ]
      })
    })
      .then((res) => {
        return res.json()
      })
      .then((payment) => {

        this.setState({ loading: false })

        // console.log("PAYMENT", payment)

        var options = {
          description: 'Kalzi Membership',
          image: 'jpg here',
          currency: currency,
          key: 'key here', // Your api key
          amount: 100,
          name: 'Hi',
          prefill: {
            email: 'email',
            name: 'name',
          },
          order_id: payment.id,
          theme: { color: 'blue' },
        }

        RazorpayCheckout.open(options).then().catch()
            
    })
    .catch()

**Comment by arya-man** [comment]
Yeah same issue here

**Comment by varsh8th** [comment]
@Nautiyalsachin @vivekshindhe & team please respond, its been bugging us since last few days

**Comment by ramprasadAnand** [comment]
Hi @varsh8th, can you pls share what async task you are performing on tap of the pay button in the UI ? The open checkout action is not async, it should open immediately on tap of button.

**Comment by arya-man** [comment]
The thing is that - the checkout works perfectly fine when it is just `RazorpayCheckout.open(options)` but the checkout does not open when a fetch request is made as in 
```
fetch(https://api.razorpay.com/v1/orders, {
method: 'POST',
headers: new Headers({
Accept: 'application/json',
'Content-Type': 'application/json',
'Authorization': Bearer Key Here

  }),
  body: JSON.stringify({
    amount: 100,
    currency: currency,
    payment: {
      capture: "automatic",
      capture_options: {
        refund_speed: "normal"
      }
    },
    transfers: [
      {
        account: linkedAccount,
        amount: 100,
        currency: currency,
        notes: {
          name: '1',
          email: '1'
        },
        linked_account_notes: [name , email]
      },
    ]
  })
})
```

and then passing the `payment.id as order_id` inside options of the checkout.
The fetch is request is made to make use of automatic capture and razorpay route. The issue is ios only and the same code works perfectly fine for android. What do you think causes this issue of the checkout not opening in ios when a fetch request is made to create an order before `RazorpayCheckout.open(options)` ? Thanks in advance!

```
var options = {
      description: 'Kalzi Membership',
      image: 'jpg here',
      currency: currency,
      key: 'key here', // Your api key
      amount: 100,
      name: 'Hi',
      prefill: {
        email: 'email',
        name: 'name',
      },
      order_id: payment.id, // not there if fetch request is not made to create razorpay order
      theme: { color: 'blue' },
    }
```

**Comment by maskedTitan** [comment]
Razorpay's APIs won't work if they're being called from the client side. They should be implemented and called in the backend and the response can be transferred to the front end.

**Comment by akshaykatale99** [comment]
Same issue for me. showing following error and sometimes it opens but most of the time throwing this error

![Screenshot 2021-07-15 at 1 15 07 PM](https://user-images.githubusercontent.com/12472207/125749561-041325ca-1617-4e72-9144-29adc2baf2ed.png)

**Comment by arya-man** [comment]
@maskedTitan Even after creating the razorpay order from the backend - the checkout is not opening. Funny thing is if I dont do the fetch call for creating order on my backend and simply call `RazorpayCheckout.open()` , the checkout opens. However if I do a post request for creating a razorpay route order on my backend - then regardless of passing `order_id in options` the checkout does not open.  Again ios specific issue and works perfectly fine on android. What do you think is the issue?

**Comment by varsh8th** [comment]
@maskedTitan @vivekshindhe @Nautiyalsachin  can you respond to @arya-man  above, he's from the same team and lead dev on the app where we are facing this issue. Its blocking lots of things for us in ios only,  as android is working fine.

We've raised multiple support requests on this since last week, support after raising tickets is basically non existent so pinging here. So can you guys take a look here and help us resolve this

**Comment by vivekshindhe** [comment]
@varsh8th We apologise for the delay. Could you send us your emailID or razorpay MID so that we can get in touch with you to resolve this sooner?

**Comment by arya-man** [comment]
Hey @vivekshindhe . Please mail at aryaman0301@gmail.com

**Comment by sahil-prakash** [comment]
> Same issue for me. showing following error and sometimes it opens but most of the time throwing this error
> 
> ![Screenshot 2021-07-15 at 1 15 07 PM](https://user-images.githubusercontent.com/12472207/125749561-041325ca-1617-4e72-9144-29adc2baf2ed.png)

I have faced similar issue in my project and have added a [PR for the fix](https://github.com/razorpay/react-native-razorpay/pull/349)

**Comment by raikk** [comment]
hii all
I have the same issue, not able to open `Razorpay` checkout screen in iOS but all working fine in android.

The issue was caused by the `react-native-loading-spinner-overlay` module, I was using spinner overlay it hide the Checkout screen model of Razorpay in iOS.

I removed the spinner overlay then all working fine.

**Comment by aditya-keri-betterhalf** [comment]
> > Same issue for me. showing following error and sometimes it opens but most of the time throwing this error
> > ![Screenshot 2021-07-15 at 1 15 07 PM](https://user-images.githubusercontent.com/12472207/125749561-041325ca-1617-4e72-9144-29adc2baf2ed.png)
> 
> I have faced similar issue in my project and have added a [PR for the fix](https://github.com/razorpay/react-native-razorpay/pull/349)

I've tried adding the code you've mentioned in the PR in `node_modules/react-native-razorpay/ios/RazorpayCheckout.m` but still facing the same issue.
Am I missing something here ?

**Comment by ket000** [comment]
It is also not working for me using iPADOS 16.5.1 on ipad and adda app. It works fine on adda.io app. It seems to be a bug with Razorpay. This is annoying. Please fix it. I reported this to Adda app support also.

**Comment by ramprasadAnand** [comment]
@ket000 are you facing this specifically on iPad OS ? Need more information on the error you are getting while opening the Payment page.

**Comment by ket000** [comment]
Yes, on ipad os. NoT sure what more info need. I have already listed the github bug. After i select hdfc for net banking the next page is completely blank. Github bug menTions specific issue with ios. Sent from my iPad ProOn Jul 10, 2023, at 6:46 AM, Ramprasad A ***@***.***> wrote:﻿
@ket000 are you facing this specifically on iPad OS ? Need more information on the error you are getting while opening the Payment page.

**Comment by Dharmesh-Dabhi** [comment]
I have same issue on ios app, its working fine in android, i tried @sahil-prakash pr code not its opening the checkout page in ios too but after payment the top navbar is gone.

yes its bug in sdk, checkout page is opening fine from app.js but not from nested component specifically bottom navigation

**Comment by ronit-reverseBits** [comment]
I was also facing the issue where the checkout was working fine on Android but not on iOS and I solved that by updating the latest React navigation library.

**Comment by hareesh7777** [comment]
facing same issue, have any solution for this ?

**Comment by RajaSah012** [comment]
## Solution: Razorpay Checkout Not Opening on iOS

### Problem
Razorpay checkout modal not appearing on iOS while working fine on Android.

### Root Cause
Loading overlays/spinners were blocking the Razorpay modal on iOS due to z-index/modal presentation conflicts.

### Fix
Clear all loading states before opening Razorpay:
```javascript
// Hide loading before Razorpay
setPayLoading(false);
if (setVerifyingPayment) setVerifyingPayment(false);

RazorpayCheckout.open(options)
  .then(async data => {
    // Show loading AFTER Razorpay closes
    if (setVerifyingPayment) setVerifyingPayment(true);
    // ... verification logic
  });
```

### Tested On
- iOS 14+ 
- Android 10+ 

This completely resolved the issue for me. Hope it helps others facing the same problem!
